### PR TITLE
DTD-874: adds stub for dm api subcontractor endpoint

### PR DESCRIPTION
**Pull request by johnregan**
Adds fixtures for `http://localhost:9111/individuals/subcontractor/idms/wmfid/blah`
Merges all DM API polling endpoints together

### CODE REVIEW

File: app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala [hunk 1]
```diff
@@ -57,4 +54,37 @@ class DebtManagementAPITestController @Inject() (
...
+    else
+      environment.getExistingFile(s"$basePath/dm/subcontractor/wmfId.json") match {
+        case None => Future.successful(NotFound("file not found"))
+        case Some(file) =>
+          val result = Source.fromFile(file).mkString.stripMargin
+          Future.successful(Ok(result))
+      }
+  }
+
+  def fieldCollectionsTemplates(): Action[FCTemplateRequest] =
```
Comment (johnregan) [inline app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala:72]: merged these dmapi requests together in the same controller

File: app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala [hunk 2]
```diff
@@ -57,4 +54,37 @@ class DebtManagementAPITestController @Inject() (
         }
   }
 
+  def getDebtDataAndDWISignals(wmfId: String): Action[AnyContent] = Action.async { request =>
+    if (appConfig.isPollingEnv)
+      pollingService.insertRequestAndServeResponse(Json.obj(), request.uri).map {
```
Comment (felixgb) [inline app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala:59]: This is inserting an empty object, is that correct?

File: app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala [hunk 3]
```diff
@@ -57,4 +54,37 @@ class DebtManagementAPITestController @Inject() (
...
+        case Some(file) =>
+          val result = Source.fromFile(file).mkString.stripMargin
+          Future.successful(Ok(result))
+      }
+  }
+
+  def fieldCollectionsTemplates(): Action[FCTemplateRequest] =
+    Action.async(parse.tolerantJson[FCTemplateRequest]) { request =>
+      if (appConfig.isPollingEnv) {
+        println(s"INSERTING Template request ${Json.prettyPrint(Json.toJson(request.body))}")
```
Comment (vinniebrice) [inline app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala:75]: Couple printlns here, are they needed?

File: app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala (hunk 1 above)
Comment (vinniebrice) [reply app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala:72]: Nice!

File: app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala (hunk 2 above)
Comment (felixgb) [reply app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala:59]: NVM looks like it's on purpose, I think I understand

Comment (johnregan) [reply app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala:59]: Yes it's only a get request so will always be empty body

File: app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (johnregan) [reply app/uk/gov/hmrc/debttransformationstub/controllers/DebtManagementAPITestController.scala:75]: my man hahah

**Pull request merged**